# **Irene's Notebook**
## Immo Eliza dataset cleaning

#### **importing libraries**

In [39]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

from IPython.display import display, HTML

# forcing pandas to display whole url
pd.set_option('display.max_colwidth', None)

# pforcing pandas to display all the columns
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
pd.set_option('display.width', None)

# Display settings used by professionals
pd.set_option('display.max_columns', 20)
pd.set_option('display.max_rows', 30)
pd.set_option('display.float_format', '{:.2f}'.format)

print(f"pandas {pd.__version__}  |  numpy {np.__version__}  ✅")


pandas 3.0.3  |  numpy 2.4.6  ✅


#### **reading CSV and extract dataset**

In [3]:
df_immovlan = pd.read_csv("data/raw/properties_cleaned.csv")
print(type(df_immovlan))
print(f'Dataset successfully red from properties_cleaned.csv')

<class 'pandas.DataFrame'>
Dataset successfully red from properties_cleaned.csv


*gonna do a copy to work on for the whole cleaning*

In [30]:
df_immo = df_immovlan.copy()

**after the cleaning, I will generate a copy of the dataset to adapt it to the investor's interests**

----------------------------------------------------------------------------------------------------------

**Deleting rows with Nan prices**  
Rows without a price are useless

In [31]:
df_immo = df_immo.dropna(subset=['price'])

**Substituting sentinel values with NaN**  
this to keep the other infos in the rows

In [32]:
df_immo.loc[df_immo['floor_number'] > 100, 'floor_number'] = pd.NA

--------------------------------------------------------------------------------------------------------------

alligning native data types

In [33]:
# mapping datatypes
type_map = {
    'price': 'float64',
    'bedrooms': 'Int64',
    'postal_code': 'Int64'
}

# vectorial conversion on the columns
df_immo = df_immo.astype(type_map)

**checking on destructive values ​​of scale**

In [34]:
df_immo.describe(include='all')

,property_id,property_type,property_subtype,price,price_type,living_area_m2,bedrooms,bathrooms,address,postal_code,...,has_terrace,total_area_m2,epc_score,region,province,nearby_city,km_from_nearby_city,is_nearby_city_prestigious,floor_number,property_url
count,11292,11292,11292,11292.00,11292,10511.00,10889.00,10013.00,9148,11292.00,...,8885.00,6213.00,9466,11292,11292,11290,11290.00,11290.00,3265.00,11292
unique,11291,2,15,NaN,1,NaN,<NA>,NaN,8602,<NA>,...,NaN,NaN,12,3,9,42,NaN,NaN,NaN,11291
top,https://immovlan.be/en/detail/residence/for-sale/7080/noirchain/vbe33505,House,residence,NaN,sell,NaN,<NA>,NaN,Dessauerplein 1,<NA>,...,NaN,NaN,C,Wallonia,Liège,Liège,NaN,NaN,NaN,https://immovlan.be/en/detail/residence/for-sale/7080/noirchain/vbe33505
freq,2,7254,5900,NaN,11292,NaN,<NA>,NaN,31,<NA>,...,NaN,NaN,1838,5473,1713,1281,NaN,NaN,NaN,2
mean,NaN,NaN,NaN,388215.93,NaN,171.73,3.06,1.39,NaN,5251.89,...,0.82,1593.60,NaN,NaN,NaN,NaN,35.78,0.05,2.71,NaN
std,NaN,NaN,NaN,357484.06,NaN,156.40,2.17,0.97,NaN,2626.68,...,0.38,5303.63,NaN,NaN,NaN,NaN,393.15,0.22,2.82,NaN
min,NaN,NaN,NaN,1.00,NaN,1.00,1.00,1.00,NaN,1000.00,...,0.00,2.00,NaN,NaN,NaN,NaN,0.00,0.00,0.00,NaN
25%,NaN,NaN,NaN,190000.00,NaN,90.00,2.00,1.00,NaN,2970.00,...,1.00,258.00,NaN,NaN,NaN,NaN,3.00,0.00,1.00,NaN
50%,NaN,NaN,NaN,299000.00,NaN,143.00,3.00,1.00,NaN,5170.00,...,1.00,622.00,NaN,NaN,NaN,NaN,8.10,0.00,2.00,NaN
75%,NaN,NaN,NaN,495000.00,NaN,205.00,4.00,2.00,NaN,7370.00,...,1.00,1302.00,NaN,NaN,NaN,NaN,16.70,0.00,3.00,NaN


filtering out not plausible values in *price* and *living_area_m2*

In [35]:
# price
cond_low_price = (df_immo['price'] <= 1)
df_immo = df_immo[~ cond_low_price]

# livingt_area_m2
cond_low_area = (df_immo['living_area_m2'] <= 1)
df_immo = df_immo[~ cond_low_area]

In [40]:
# df_immo.describe(include='all')
# trasposted chart - columns are visualizable as rows

# forces HTML horizontal rendering
display(HTML(df_immo.describe().to_html()))

,price,living_area_m2,bedrooms,bathrooms,postal_code,latitude,longitude,building_year,furnished,has_garage,parking_count,has_elevator,facades,floors_total,has_garden,garden_area_m2,has_terrace,total_area_m2,km_from_nearby_city,is_nearby_city_prestigious,floor_number
count,11283.00,10502.00,10880.00,10006.00,11283.00,11281.00,11281.00,6649.00,6686.00,4889.00,3701.00,7983.00,8475.00,5057.00,8105.00,2468.00,8880.00,6207.00,11281.00,11281.00,3265.00
mean,388359.42,171.87,3.06,1.39,5251.81,50.55,4.72,1970.24,0.06,1.00,1.90,0.28,2.94,3.37,0.72,1209.32,0.82,1592.11,35.80,0.05,2.71
std,357572.80,156.39,2.17,0.97,2627.55,2.82,2.90,48.23,0.24,0.00,31.18,0.45,0.88,3.00,0.45,5506.93,0.38,5303.30,393.31,0.22,2.82
min,3048.00,11.00,1.00,1.00,1000.00,3.04,2.51,1500.00,0.00,1.00,1.00,0.00,1.00,1.00,0.00,1.00,0.00,2.00,0.00,0.00,0.00
25%,190000.00,90.00,2.00,1.00,2970.00,50.46,4.07,1950.00,0.00,1.00,1.00,0.00,2.00,2.00,0.00,106.75,1.00,259.00,3.00,0.00,1.00
50%,299000.00,143.00,3.00,1.00,5170.00,50.78,4.47,1975.00,0.00,1.00,1.00,0.00,3.00,3.00,1.00,331.50,1.00,622.00,8.10,0.00,2.00
75%,495000.00,205.00,4.00,2.00,7370.00,51.03,5.31,2008.00,0.00,1.00,1.00,1.00,4.00,4.00,1.00,899.25,1.00,1302.00,16.70,0.00,3.00
max,13000000.00,3990.00,100.00,23.00,9991.00,51.47,51.35,2028.00,1.00,1.00,1885.00,1.00,8.00,30.00,1.00,150000.00,1.00,178367.00,6681.30,1.00,26.00


- 3048 lowest price to check
- 11 m2 lowest living area to check

In [41]:
# check lowest price url
suspect_url_price = df_immo.loc[df_immo['price'] == 3048.00, 'property_url']
print(f"Suspect url for {df_immo['price'].min()}:")
print(suspect_url_price)
# check lowest living_area_m2 url

Suspect url for 3048.0:
10077    https://immovlan.be/en/detail/apartment/for-sale/8400/oostende/rwc42370
Name: property_url, dtype: str


Label 10077 property is probably for rent.  

I consider a realistic minimum limit for sale (Floor price): 50.000